Assertion mistake in ParallelCompactionsST

### Description / Background

In the following previous issue we updated the assertions in ParallelCompactionsST:
- https://github.com/gchq/sleeper/issues/7423

There was a mistake in this change that causes the test to always fail.

### Steps to reproduce

1. Run ParallelCompactionsST
2. See error

### Expected behaviour

The test should pass if the jobs run in a way that is described by the assertions.

### Technical Notes / Implementation Details

This is a problem with the assertion, in JobsFinishedStatistics.isAllFinishedWithMinimumJobsAndMaxRetries.

### Screenshots/Logs

Test output:
```
java.lang.AssertionError: 
Expecting actual:
  JobsFinishedStatistics{numJobs=73728, numFinishedJobs=73728, numJobRuns=73730, numFinishedJobRuns=73728, averageRowRate=AverageRowRate{runCount=73728, rowsRead=33689357, rowsWritten=33689357, totalDuration=PT5H45M55.092S, rowsReadPerSecond=1623.1851441564315, rowsWrittenPerSecond=1623.1851441564315, averageRunRowsReadPerSecond=1838.2504559543797, averageRunRowsWrittenPerSecond=1838.2504559543797}}
to match 'all jobs finished with at most 2 retries' predicate.
```

Here numJobs and numFinishedJobs were equal, numJobRuns was two higher than numJobs, numFinishedJobRuns was equal to numFinishedJobs. This means there were exactly two retries, so the assertion should have passed.

--- a/java/system-test/system-test-dsl/src/main/java/sleeper/systemtest/dsl/reporting/JobsFinishedStatistics.java
+++ b/java/system-test/system-test-dsl/src/main/java/sleeper/systemtest/dsl/reporting/JobsFinishedStatistics.java
@@ -74,7 +74,7 @@ public class JobsFinishedStatistics {
         if (numJobs < minimumJobs) {
             return false;
         }
-        if (numJobRuns > minimumJobs + maxRetries) {
+        if (numJobRuns > numJobs + maxRetries) {
             return false;
         }
         return isAllFinished(numJobs);

